src/main/java/seedu/contax/model/AddressBook.java

```diff
@@ -47,14 +47,22 @@ public AddressBook(ReadOnlyAddressBook toBeCopied) {
     //// list overwrite operations
 
     /**
-     * Replaces the contents of the person list with {@code persons}.
+     * Replaces the contents of the person list with {@code persons} and adds any tags that does not exist in the tag
+     * list.
      * {@code persons} must not contain duplicate persons.
      */
     public void setPersons(List<Person> persons) {
+        persons.forEach(this::addTagsNotInList);
```

Comment:
Add a `requireNonNull` check for `persons`, since this is a potential source of `NullPointerException`.